garden-backend-service/src/api/routes/modal/modal_apps.py

```diff
@@ -278,6 +196,15 @@ async def delete_modal_app(
             detail=f"No Modal App found with id {id}.",
         )
 
+    _raise_if_undeletable(modal_app, user, log)
+
+    await db.delete(modal_app)
+    await db.commit()
+    log.info("Deleted Modal App from database")
+    return {"detail": f"Successfully deleted garden with id {id}."}
+
+
+def _raise_if_undeletable(modal_app, user, log):
     if modal_app.owner.identity_id != user.identity_id:
         log.info("Failed to delete Modal App (not owned by user)")
         raise HTTPException(
```

Comment:
This isn't brought in by your PR, but just noticing that the detail in this exception seems to indicate something different from the condition that raised it